src/internal/proc.js

```diff
@@ -600,6 +601,21 @@ export default function proc(
     cb(!!mainTask.isCancelled)
   }
 
+  function runFlushEffect(channel, cb) {
+    const bufferedItems = []
+    for (;;) {
+      let exposedItem
+      channel.take(item => exposedItem = item)
```

Comment:
if the buffer is empty, this callback wont be called until the next item is put into the channel. The saga will remain blocked until then